### DISCUSSION TITLE
Extension activation failed: "connect ETIMEDOUT

### DISCUSSION THREAD

**Discussion by itimoreo** [community/community#113550] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

Hi, Today I launched VS Code, and copilot told me this: "Extension activation failed: "connect ETIMEDOUT 140.82.121.5:443"
I have no proxy, no firewall, no VPN. How to do?

**Comment by Akash1134** [comment]
Hi @itimoreo ,

I suspect this may be related to the [deprecation of Copilot Chat API endpoints](https://github.blog/changelog/2024-01-18-copilot-january-18th-update/) so please review if you haven't so far:

> To ensure Copilot Chat continues working, please update to the most recent version of the Copilot extension and ensure your firewall and network settings allow communication to https://api.githubcopilot.com.

Additionally, the error `ETIMEDOUT` typically occurs when you're using a program that intercepts network connections, even if there are no problems with the network itself. Please note that GitHub Copilot may reject some custom certificates. For more information, see [Troubleshooting certificate-related errors](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot#troubleshooting-certificate-related-errors).

Let us know if that helps.

**Reply by itimoreo** [reply]
I have followed the recommended steps to troubleshoot the persistent ETIMEDOUT errors I've been experiencing with the GitHub Copilot extension in Visual Studio Code, but unfortunately, the issue remains unresolved. To summarize the steps I've taken:

GitHub Copilot Update: I have ensured that the GitHub Copilot extension is up to date.
Firewall Configuration: I have verified that Visual Studio Code is allowed through the Windows Firewall for both private and public networks.
Internet Properties SSL/TLS Settings: I have checked the SSL and TLS settings in the Internet Properties, ensuring that the recommended options are selected (TLS 1.2 and TLS 1.3).
Despite these,  I am still facing connectivity issues with the GitHub Copilot extension. There are no active VPNs, proxies, antivirus software, or enterprise-grade firewalls that could be interfering with the connection.

Could there be any other underlying factors that might cause these timeout errors? 

Also I tried to Uninstall et Re-install VS Code.

**Comment by itimoreo** [comment, marked as answer]
The problem is solved, i don't know how but it's solve

**Reply by andrefelipe18** [reply]
I started having the same problem on Sunday, and so far I haven't been able to solve it, I've tried everything you've said above

**Reply by legasrossini** [reply]
@andrefelipe18  conseguiu?

**Reply by andrefelipe18** [reply]
No :'(

**Reply by RunYouCleverBoy** [reply]
No solution here, too. I found some [post](https://github.com/orgs/community/discussions/64334) that says "add 
```
140.82.113.5    api.github.com
```
to the hosts. Now I can ping `api.github.com` but I get an
```
[error] request to https://api.github.com/user failed, reason: read ECONNRESET
```
😢 

Let alone that it's not secure, becaue it's just trying some IP from a forum online.

I tried to logout from GitHub in VSCode. Now I can't use Copilot, can't push anything to Git via VSCode. Dead as disco, never seen alive again.
I suspect that the auth server is dead because on Android Studio it shows this:

<img width="463" alt="image" src="https://github.com/community/community/assets/14019958/48bf6eba-e1cb-4dd8-a3ac-f124d7005c69">

When I use "Login via GitHub", its dead.
But even when I pick "Use token", and generate a brand new token with permissions that even the Roman Emperors didn't have - still nothing happens, hangs on login and fails.

**Comment by srfreitas97** [comment]
im having the same problem here... did you guys did any troubleshooting or the issue resolved itself?

**Comment by andrefelipe18** [comment]
Same problem here :/

**Comment by rpjax** [comment]
Same problem here guys...

**Comment by andrefelipe18** [comment]
Guys... I opened vscode and the copilot still didn't work, but I left it open for a while and went out to do other things, and when I came back to the computer the copilot had started by itself

**Reply by andrefelipe18** [reply]
![image](https://github.com/community/community/assets/96439642/fe07f4e2-6a5e-4eb3-9971-292f64801b7c)
